src/department/department.controller.ts

```diff
@@ -38,6 +49,19 @@ export class DepartmentController {
   ): Promise<DepartmentEntity> {
     return this.departmentServise.createDepartment(createDepartmentDto);
   }
+  @Post(":id/employees")
+  @UseGuards(AuthGuard)
+  @UsePipes(new ValidationPipe())
+  async createUserInDepartment(
+    @Param("id") id: number,
+    @Body() createEmployeeDto: CreateEmployeeDto
+  ): Promise<EmployeeEntity> {
+    const currentDepartment = await this.departmentServise.getById(id);
+    return this.employeeSevice.createEmployee(
+      currentDepartment,
+      createEmployeeDto
+    );
```

Comment:
Creation an employee in department is more up to the employeeService rather than department controller. Just move this logic there and rename the method to createEmployeeInDepartment

Reply:
I can move from department.controller, but then how do I save the URL "/api/v2/departments/:id/employees" when I create an employee.

Reply:
I try to make these endpoints   
/api/v2/departments (POST)
/api/v2/departments (GET)
/api/v2/departments/:id(GET)
/api/v2/departments/:id (DELETE)
/api/v2/departments/:id (PATCH)
/api/v2/departments/:id/employees (POST)

/api/v2/employees/:id (GET)
/api/v2/employees/:id (PATCH)
/api/v2/employees/:id  (DELETE)